onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorDynamicQuantizeMatMul.cpp

```diff
@@ -40,16 +40,21 @@ class DmlOperatorDynamicQuantizeMatMul : public DmlOperator
                 kernelCreationContext.GetTensorShapeDescription().GetOutputTensorShape(0)
             );
         }
+        MLOperatorTensorDataType ADatatype = kernelCreationContext.GetInputEdgeDescription(OnnxInputIndex::A).tensorDataType;
         MLOperatorTensorDataType BDatatype = kernelCreationContext.GetInputEdgeDescription(OnnxInputIndex::B).tensorDataType;
 
         std::vector<uint32_t> ATensorShape = kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(OnnxInputIndex::A);
+        std::vector<uint32_t> BTensorShape = kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(OnnxInputIndex::B);
         std::vector<uint32_t> ExpectedAScaleTensorShape = {1, 1, 1, 1};
         std::vector<uint32_t> ExpectedAZeroPointTensorShape = {1, 1, 1, 1};
+        auto& outputSizes = m_outputTensorDescs[0].GetSizes();
+        auto AShapeBroadcasted = std::array<uint32_t, 4> { outputSizes[0], outputSizes[1], ATensorShape.rbegin()[1], ATensorShape.rbegin()[0] };
+        auto BShapeBroadcasted = std::array<uint32_t, 4> { outputSizes[0], outputSizes[1], BTensorShape.rbegin()[1], BTensorShape.rbegin()[0] };
```

Comment:
The docs for [`com.microsoft.DynamicQuantizeMatMul`](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftdynamicquantizematmul) say "N-dimensional matrix A". Do we handle 5D correctly? It's probably okay if we don't if only 4D is interesting, but then I'd `ML_CHECK_VALID_ARGUMENT(outputSizes.size() == 4);` earlier to verify 4D exactly. That will be a clearer error message than a shape mismatch error.